### ISSUE TITLE
Altering - cluster-altering app shows errors after upgrading Rancher

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#21416]
Steps:
- run Rancher: v2.2.3 container
- as admin, enable Github auth
- log in as Github user user1
- add the following three clusters: DO single-node, DO 3-node, and GKE
- adds cluster notifier and cluster alerting for all three clusters
- upgrade Rancher to v2.2.5-rc12

Results:
- for all three clusters, the cluster altering still work
- for all three clusters, adding a new alter group and new rule also works
- for all three clusters, the app `cluster-alerting` is stuck in `installing` state, like this:
<img width="1463" alt="from v2 2 3 to v2 2 5-rc12" src="https://user-images.githubusercontent.com/6218999/60932137-7f2a6600-a271-11e9-95f5-2754e7563f48.png">

- the app `cluster-alerting` in one cluster shows the following error
```
Failed to install app cluster-alerting. Error: UPGRADE FAILED: the server is currently unable to handle the request (get configmaps)
```
<img width="1479" alt="Screen Shot 2019-07-09 at 5 47 48 PM" src="https://user-images.githubusercontent.com/6218999/60932185-b13bc800-a271-11e9-9b6e-9bdc6c4f3c87.png">

- in another two clusters, it shows this error
```
Failed to install app cluster-alerting. Error: stat /rancher-monitoring: no such file or directory
```
<img width="1456" alt="from v2 2 3 to v2 2 5-rc12 " src="https://user-images.githubusercontent.com/6218999/60932219-d6c8d180-a271-11e9-96c9-4f6fc6a3fc97.png">

One more test:
- add a new DO single-node cluster in the same setup 
- enable the cluster altering 

Results:
- the cluster altering works

**Comment by sangeethah** [comment]
To get the cluster-altering to active state , "Click upgrade and save".

**Comment by cjellick** [comment]
most straight foward steps to repro:

1. Start rancher 2.2.3
2. Add 3~5 clusters in the cluster.
3. Add notifier in each cluster
4. Edit cluster alert group and add a recipient
5. Check if alerting is enabled by looking at apps in system project
6. Upgrade rancher to 2.2.5-rc12
7. Check the status of monitoring-operator apps 


Cannot repro if `CATTLE_DEV_MODE` is used to disable jailing.

**Comment by loganhz** [comment]
Some error message might be useful. Not sure this is a Jail issue or helm controller issue.

```
2019/07/10 13:14:58 [ERROR] AppController p-9v8p8/cluster-alerting [helm-controller] failed with : start helm template --notes failed. : fork/exec /usr/bin/helm: no
such file or directory
2019/07/10 13:14:58 [ERROR] AppController p-9v8p8/cluster-alerting [helm-controller] failed with : start helm template failed. : fork/exec /usr/bin/helm: no such fil
e or directory
2019/07/10 13:14:58 [ERROR] AppController p-9v8p8/cluster-alerting [helm-controller] failed with : start helm template failed. : fork/exec /usr/bin/helm: no such fil
e or directory
2019/07/10 13:14:59 [ERROR] AppController p-9v8p8/cluster-alerting [helm-controller] failed with : failed to install app cluster-alerting. Error: stat /rancher-monitoring: no such file or directory
2019/07/10 13:15:58 [WARNING] START TILLER ERROR fork/exec /usr/bin/tiller: no such file or directory
2019/07/10 13:15:58 [ERROR] got error while stopping tiller, error message: fork/exec /usr/bin/tiller: no such file or directory
```

```
2019/07/10 13:37:44 [ERROR] AppController p-f9dng/monitoring-operator [helm-controller] failed with : failed to install app monitoring-operator. Error: stat /rancher
-monitoring: no such file or directory
2019/07/10 13:38:59 [ERROR] AppController p-nwmcq/cluster-alerting [helm-controller] failed with : failed to install app cluster-alerting. Error: stat /rancher-monit
oring: no such file or directory
2019/07/10 13:38:59 [ERROR] AppController p-f9dng/cluster-alerting [helm-controller] failed with : failed to install app cluster-alerting. Error: stat /rancher-monit
oring: no such file or directory
2019/07/10 13:39:59 [ERROR] AppController p-nwmcq/rancher-logging [helm-controller] failed with : failed to install app rancher-logging. Error: stat /rancher-logging
: no such file or directory
```

**Comment by cjellick** [comment]
merged. will be available in next v2.2-head build

**Comment by jiaqiluo** [comment]
The bug fix is validated in the upgrade path from Rancher v2.2.3 to  v2.2-head 8268cd5f

Did the same steps as the issue describes, and the cluster altering works as expected in all three clusters after upgrading Rancher; the cluster altering also works in the newly-added cluster.
